rust/src/delta.rs

```diff
@@ -1153,6 +1180,19 @@ impl<'a> DeltaTransaction<'a> {
     }
 }
 
+fn log_bytes_from_actions(actions: &[Action]) -> Result<String, serde_json::Error> {
```

Comment:
I find it awkward to have a private function just floating in the top level namespace when it's really only something needed/used by the functions within the `DeltaTransaction`. 

I suggest moving this up in to there.

Reply:
This is a super interesting style point. I always have an internal battle in cases like this. The rule I've been following (privately made rule) up to now has been - "if my fn doesn't have a self ref - it should go into an outer scope since it does not depend on self state" - implication - all struct "methods" should be true methods that have an actual dependency on self state.

I don't disagree with your opinion necessarily, but I'd love to discover a hard rule to base placement decisions like this on rather than a smell which is a bit more ambiguous.

Reply:
If wonder if we were to create a `mod` for `delta_transaction` and this `fn` lived in that `mod` but still out side of the `impl DeltaTransaction` if this might be more palatable to both of us.